core-primitives/top-pool-author/src/author.rs

```diff
@@ -204,6 +204,44 @@ where
 			),
 		}
 	}
+
+	fn remove_top(
```

Comment:
Removed this function from the trait and changed it, so it only removes a single operation at a time (as the name suggests). And it now correctly returns an error when the decoding fails (before we also returned a Result, but it was always `Ok`, and on `decode` we called `unwrap()`, which made little sense).